File: src/main/scala/s/decorator/inheritance/StringDisplay.scala
package s.decorator.inheritance

final class StringDisplay(private val string: String) extends Display {
  override def columns: Int = string.length

  override def rows = 1

  override def rowText(row: Int): String = {
    if (row != 0) throw new IndexOutOfBoundsException
    string
  }
}

File: src/main/java/j/decorator/inheritance/StringDisplay.java
package j.decorator.inheritance;

public final class StringDisplay extends Display {

  private final String string;

  StringDisplay(String string) {
    this.string = string;
  }

  @Override
  public int getColumns() {
    return string.length();
  }

  @Override
  public int getRows() {
    return 1;
  }

  @Override
  public String getRowText(int row) {
    if (row != 0) {
      throw new IndexOutOfBoundsException();
    }
    return string;
  }
}
